internal/provider/kubernetes/controller.go

```diff
@@ -1273,18 +1273,33 @@ func (r *gatewayAPIReconciler) processBackendTLSPolicies(
 }
 
 // processBackends adds Backends to the resourceTree
-func (r *gatewayAPIReconciler) processBackends(ctx context.Context, resourceTree *resource.Resources) error {
+func (r *gatewayAPIReconciler) processBackends(ctx context.Context, resourceTree *resource.Resources, resourceMappings *resourceMappings) error {
```

Comment:
or can we simply remove this function? at the end of the day, all referenced backends are properly added later in processBackendRefs. @arkodg

Reply:
If I remember correctly, the main purpose was to add the backends so that we can update their status to reflect the situation where the backend toggle is not enabled, regardless of them being referenced by any backendref. 

We can change the behavior to reflect that in the xRoute/Policy resource status instead. I think that the UX change is not very significant here.